### PR TITLE
added a public var to extract the IP address of the SocketAddress

### PR DESCRIPTION
**Pull request by Lupurus**
see #1190

### CODE REVIEW

**Comment by Lupurus** [comment]
Oh sorry! I'm totally new to UnitTesting (shame on me). I hope, the new commit should do the job.

File: Tests/NIOTests/SocketAddressTest.swift [hunk 1]
```diff
@@ -65,6 +65,13 @@ class SocketAddressTest: XCTestCase {
         XCTAssertEqual(s, sampleString,
                        "Address description is way below our expectations 😱")
     }
+	
+	func testIPAddressWorks() throws {
+		let sa = try! SocketAddress(ipAddress: "127.0.0.1", port: 12345)
+		XCTAssertEqual("127.0.0.1", sa.ipAddress)
+		let sa6 = try! SocketAddress(ipAddress: "::1", port: 12345)
+		XCTAssertEqual("::1", sa6.ipAddress)
+	}
```
Comment (Lukasa) [inline Tests/NIOTests/SocketAddressTest.swift]: Can we also add a test that the return value is nil for unix domain sockets? Then we'll have covered all three branches.

**Comment by Lukasa** [comment]
Heh, no problem! Everyone is new to unit testing at some stage, so there is no need to apologise.

**Comment by Lupurus** [comment]
> Heh, no problem! Everyone is new to unit testing at some stage, so there is no need to apologise.

:D but still thank you for your patience (and now I learned sth. thats really cool)

File: Sources/NIO/SocketAddresses.swift [hunk 2]
```diff
@@ -129,6 +129,22 @@ public enum SocketAddress: CustomStringConvertible {
...
+			var mutAddr = addr.address.sin_addr
+			// this uses inet_ntop which is documented to only fail if family is not AF_INET or AF_INET6 (or ENOSPC)
+			return try! descriptionForAddress(family: AF_INET, bytes: &mutAddr, length: Int(INET_ADDRSTRLEN))
+		case .v6(let addr):
+			var mutAddr = addr.address.sin6_addr
+			// this uses inet_ntop which is documented to only fail if family is not AF_INET or AF_INET6 (or ENOSPC)
+			return try! descriptionForAddress(family: AF_INET6, bytes: &mutAddr, length: Int(INET6_ADDRSTRLEN))
+		case .unixDomainSocket(_):
+			return nil
+		}
```
Comment (Lukasa) [inline Sources/NIO/SocketAddresses.swift]: Here's a really minor nit: can you use 4-space indentation? That's more consistent with the rest of the project.

**Comment by Lukasa** [comment]
@swift-nio-bot test this please

**Comment by Lukasa** [comment]
Ah, the sanity check script has flagged another issue. Can you run `ruby scripts/generate_linux_tests.rb` as well, and commit the results?

**Comment by weissi** [comment]
@swift-nio-bot test this please

**Comment by weissi** [comment]
@swift-nio-bot test this please
